### DISCUSSION TITLE
How is readonly access to a Kafka-Cluster implemented?

### DISCUSSION THREAD

**Discussion by jfitschen** [tchiotludo/akhq#1494] [Q&A]
Hi,
I couldn'd find the method to connect to a Kafka-Cluster without (as it seems) committing the messages in a topic.
Is there a way to get an "readonly" acces to a topic and read messages multiple times?
Thanks a lot in advance
Jürgen

**Comment by tchiotludo** [comment]
You need to setup [authentifications](https://akhq.io/docs/configuration/authentifications/)

**Reply by jfitschen** [reply]
Hi,
thanks for the quick answer. I was looking for a mechanism to get readonly access for my own application. AKHQ seems to use a good solution for accessing messages in a topic, but I couldn't figure out how to do this myself in a spring or java app.
Sorry for the lack of clarity

**Reply by tchiotludo** [reply]
it's clearly not the good of the application, use the default [java kafka driver ](https://docs.confluent.io/kafka-clients/java/current/overview.html)

**Reply by jfitschen** [reply]
the normal way of reading messages from a topic includes the committing in the offsets in the internal topic __offset_commit. AKHQ doesn't seem to use this mechanism, or am I wrong?

**Reply by tchiotludo** [reply]
yes and it's handle by java driver, just need to remove the commit of offset in the java implementation you use

**Reply by jfitschen** [reply]
ah, ok, I see. Thanks!!
